test/utils/deployment.ts

```diff
@@ -0,0 +1,121 @@
+        yarnStartProcess.stdout.on('data', chunk => {
+            neatlyPrintChunk('[0x-api]', chunk);
+        });
+        yarnStartProcess.stderr.on('data', chunk => {
+            neatlyPrintChunk('[0x-api | error]', chunk);
+        });
+    }
+    // Wait for the API to boot up
+    // HACK(jalextowle): This should really be replaced by log-scraping, but it
+    // does appear to work for now.
+    await sleepAsync(5000); // tslint:disable-line:custom-no-magic-numbers
+}
+
+/**
+ * Tears down the old 0x-api instance.
+ * @param loggingConfig Whether or not the logs from the teardown functions should
+ *        be printed.
+ */
+export async function teardownApiAsync(loggingConfig: LoggingConfig = {}): Promise<void> {
+    if (!yarnStartProcess) {
+        throw new Error('There is no 0x-api instance to tear down');
+    }
+    yarnStartProcess.kill();
+    await teardownDependenciesAsync(loggingConfig.shouldPrintDependencyLogs || false);
+}
+
+/**
+ * Sets up 0x-api's dependencies.
+ * @param shouldPrintLogs Whether or not the logs from `docker-compose up`
+ *        should be printed.
+ */
+export async function setupDependenciesAsync(shouldPrintLogs: boolean = false): Promise<void> {
+    const up = spawn('docker-compose', ['up'], {
+        cwd: apiRootDir,
+        env: {
+            ...process.env,
+            ETHEREUM_RPC_URL: 'http://ganache:8545',
+            ETHEREUM_CHAIN_ID: '1337',
+        },
+    });
+    if (shouldPrintLogs) {
+        up.stdout.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose up]', chunk);
+        });
+        up.stderr.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose up | error]', chunk);
+        });
+    }
+    // Wait for the dependencies to boot up.
+    // HACK(jalextowle): This should really be replaced by log-scraping, but it
+    // does appear to work for now.
+    await sleepAsync(10000); // tslint:disable-line:custom-no-magic-numbers
+}
+
+/**
+ * Tears down 0x-api's dependencies.
+ * @param shouldPrintLogs Whether or not the logs from `docker-compose down`
+ *        should be printed.
+ */
+export async function teardownDependenciesAsync(shouldPrintLogs: boolean = false): Promise<void> {
+    const down = spawn('docker-compose', ['down'], {
+        cwd: apiRootDir,
+    });
+    if (shouldPrintLogs) {
+        down.stdout.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose down]', chunk);
+        });
+        down.stderr.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose down | error]', chunk);
+        });
+    }
+}
+
+function neatlyPrintChunk(prefix: string, chunk: Buffer): void {
+    const data = chunk.toString().split('\n');
+    data.filter((datum: string) => datum !== '').map((datum: string) => {
+        log.log(prefix, datum.trim());
+    });
+}
+
+async function sleepAsync(duration: number): Promise<void> {
+    return new Promise<void>(resolve => {
+        setTimeout(resolve, duration);
+    });
+}
```

Comment:
deployment is pretty sweet 🙌  any idea how much time processes take to start up/tear down this way?

One thing I see needing is a way to delete all orders from the postgres cache between each test. We don't necessarily need to wipe all the deps, just clear the saved orders. Would this be an appropriate place for a function like that?

Reply:
The sleeps provide a hard cap on how long it takes. As per Fabio's comment below, I am now deleting the database and mesh records, which added several seconds. It's looking like around ~15 - 20s for setup and very few (1 - 4s at most) for teardown.

I haven't really worked on this at all since I'm just trying to get the framework out the door so we can start to write some tests, but I will look into scraping the logs this afternoon.